sdks/java/extensions/sketching/src/test/java/org/apache/beam/sdk/extensions/sketching/TDigestQuantilesTest.java

```diff
@@ -0,0 +1,179 @@
+            .apply(TDigestQuantiles.globally().withCompression(compression))
+            .apply(ParDo.of(new RetrieveQuantiles(0.25, 0.5, 0.75, 0.99)));
+
+    PAssert.that("Verify Accuracy", col).satisfies(new VerifyAccuracy());
+    tp.run();
+  }
+
+  @Test
+  public void perKey() {
+    PCollection<KV<Double, Double>> col = tp.apply(Create.of(stream))
+            .apply(WithKeys.<Integer, Double>of(1))
+            .apply(TDigestQuantiles.<Integer>perKey().withCompression(compression))
+            .apply(Values.<MergingDigest>create())
+            .apply(ParDo.of(new RetrieveQuantiles(0.25,  0.5, 0.75, 0.99)));
+
+    PAssert.that("Verify Accuracy", col).satisfies(new VerifyAccuracy());
+
+    tp.run();
+  }
+
+  @Test
+  public void testCoder() throws Exception {
+    MergingDigest tDigest = new MergingDigest(1000);
+    for (int i = 0; i < 10; i++) {
+      tDigest.add(2.4 + i);
+    }
+
+    Assert.assertTrue("Encode and Decode", encodeDecodeEquals(tDigest));
+  }
+
+  @Rule
+  public ExpectedException thrown = ExpectedException.none();
+
+  @Test
+  public void testMergeAccum() {
+      Random rd = new Random(1234);
+      List<MergingDigest> accums = new ArrayList<>();
+      for (int i = 0; i < 3; i++) {
+          MergingDigest std = new MergingDigest(100);
+          for (int j = 0; j < 1000; j++) {
+              std.add(rd.nextDouble());
+          }
+          accums.add(std);
+      }
+      TDigestQuantilesFn fn = TDigestQuantilesFn.create(100);
+      MergingDigest res = fn.mergeAccumulators(accums);
+  }
+
+  private <T> boolean encodeDecodeEquals(MergingDigest tDigest) throws IOException {
+    ByteArrayOutputStream baos = new ByteArrayOutputStream();
+    MergingDigestCoder tDigestCoder = new MergingDigestCoder();
+
+    tDigestCoder.encode(tDigest, baos);
+    byte[] bytes = baos.toByteArray();
+
+    ByteArrayInputStream bais = new ByteArrayInputStream(bytes);
+    MergingDigest decoded = tDigestCoder.decode(bais);
+
+    boolean equal = true;
+    // the only way to compare the two sketches is to compare them centroid by centroid.
+    // Indeed, the means are doubles but are encoded as float and cast during decoding.
+    // This entails a small approximation that makes the centroids different after decoding.
+    Iterator<Centroid> it1 = decoded.centroids().iterator();
+    Iterator<Centroid> it2 = tDigest.centroids().iterator();
+
+    for (int i = 0; i < decoded.centroids().size(); i++) {
+      Centroid c1 = it1.next();
+      Centroid c2 = it2.next();
+      if ((float) c1.mean() != (float) c2.mean() || c1.count() != c2.count()) {
+        equal = false;
+        break;
+      }
+    }
+    return equal;
+  }
+
+  static class RetrieveQuantiles extends DoFn<MergingDigest, KV<Double, Double>> {
+    private final double quantile;
+    private final double[] otherQ;
+
+    public RetrieveQuantiles(double q, double... otherQ) {
+      this.quantile = q;
```

Comment:
Why do you need two variables: why not just a single `double[]` ? It doesn't seem that `q` is special.

Reply:
If there is only one quantile it would not work. It was for better adaptability of the code, for instance if someone want to try something in the tests, but it is not really important, I can change if you want...